AdditionalWeaponSlots.cpp

```diff
@@ -73,10 +96,32 @@ HOOK_METHOD_PRIORITY(WeaponControl, RenderSelfAiming, 9999, () -> void)
         {
             if (weapon->blueprint->type == 2)
             {
-                if (weapon->targets.size() >= 2)
+                if (weapon->targets.size() >= 2 && weapon->blueprint->length > 1)
                 {
                     RenderBeamAiming(weapon->targets[1], weapon->targets[0], weapon->autoFiring);
                 }
+                
+                else if(weapon->targets.size() >= 1 && weapon->blueprint->length == 1)
+                {
+                    GL_Primitive *prim;
```

Comment:
Code duplication; maybe we should turn this into a function/method like with RenderBeamAiming.

Reply:
Resolved code duplication